### PR TITLE
Sphinx no longer supports Python 2.6

### PR DESCRIPTION
**Pull request by adamjstewart**
The last half-dozen PRs are failing their documentation tests because pip fails to install sphinx for Python 2.6. We may also need #4243 to get them working again.

### CODE REVIEW

**Comment by adamjstewart** [comment]
@tgamblin Any idea what's up with this?
```
Extension error:
Could not import extension sphinxcontrib.programoutput (exception: No module named programoutput)
make: *** [html] Error 1
```
I'm not sure why this happened all of a sudden. Maybe we need to install setuptools too? I was also thinking about adding a newer version of programoutput.

**Comment by adamjstewart** [comment]
Or we could `pip install sphinxcontrib-programoutput`.

**Comment by tgamblin** [comment]
It's vendored in the docs directory under `exts`, so this is confusing to me too.  Want me to look at it?

**Comment by adamjstewart** [comment]
Yeah, I've been staring at it for a while now. Can't reproduce it locally. I can't imagine that `setuptools` isn't installed on the system.

**Comment by adamjstewart** [comment]
@tgamblin Can you take a look at the last commit? I'm pretty sure our documentation can't be built in parallel. See https://github.com/NextThought/sphinxcontrib-programoutput/issues/25 for discussion. If you apply https://github.com/NextThought/sphinxcontrib-programoutput/pull/26 and try building in parallel, you'll find that there are dozens of error messages. I think the problem is that our documentation includes references to other files, and if those files don't exist at the time of the creation, they raise an error.

If we could get our documentation working properly in parallel, that would be pretty awesome. In serial (which is how we've always built things, thanks to `programoutput`, the build takes 2.5 - 3 minutes. In parallel, even on a 4 core machine, we're under 30 seconds flat.

**Comment by adamjstewart** [comment]
Ok, gave up on trying to get Sphinx to find programoutput on its own. Let's just let Travis install it. We don't vendor Sphinx with Spack, so why vendor `sphinxcontrib-programoutput`?

If this last commit doesn't fix all of the Travis error messages, I'm out of ideas.

**Comment by tgamblin** [comment]
@adamjstewart: I was able to reproduce this locally by updating sphinx.  The issue is that `sphinxcontrib` seems to be on the default `sys.path` now, and *that* `sphinxcontrib` ends up loaded before ours.  Maybe the directory comes with newer versions of `sphinx`?

Here's a reproducer for the same issue.  make a directory like this:

```
1/
    a/
        __init__.py
        b/
            __init__.py
2/
    a/
        __init__.py
        c/
            __init__.py
test.py
```

where `test.py` looks like this:

```python
#!/usr/bin/env python

import sys

sys.path.append('1')
sys.path.append('2')

import a
import a.b
import a.c
```

Running it gives:

```
$ ./test.py
Traceback (most recent call last):
  File "./test.py", line 11, in <module>
    import a.c
ImportError: No module named c
```

If you tweak `conf.py` a bit in the `docs` dir, you can see that the same thing is happening:

```python
...
sys.path.insert(0, os.path.abspath('exts'))                                                                                                                                                                                                 
                                                                                                                                                                                                                                            
import sphinxcontrib                                                                                                                                                                                                                        
print(sphinxcontrib.__path__)   
```

Then: 

```
$ make
sphinx-build -j 8 -b html -d _build/doctrees  -E . _build/html
Running Sphinx v1.6.1
sphinxcontrib
['/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/sphinxcontrib']
...

$ ls /usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/sphinxcontrib
websupport/
```

Then, back in `conf.py`:

```python
...
sys.path.insert(0, os.path.abspath('exts'))                                                                                                                                                                                                 
                                                                                                                                                                                                                                            
import sphinxcontrib                                                                                                                                                                                                                        
import sphinxcontrib.websupport                                                                                                                                                                                                             
import sphinxcontrib.programoutput
...
```

The first two imports succeed but not the last:

```
$ make
sphinx-build -j 8 -b html -d _build/doctrees  -E . _build/html
Running Sphinx v1.6.1

Exception occurred:
  File "conf.py", line 56, in <module>
    import sphinxcontrib.programoutput
ImportError: No module named programoutput
The full traceback has been saved in /var/folders/0s/q_y0zhfj6xdd5n7rn780zz6h001qr7/T/sphinx-err-3gPAmq.log, if you want to report the issue to the developers.
```

I think the right thing to do is to just install `programoutput` with `pip`.

**Comment by adamjstewart** [comment]
Should we drop the vendored version altogether then?

**Comment by tgamblin** [comment]
Yes

File: lib/spack/docs/contribution_guide.rst [hunk 1]
```diff
@@ -189,6 +189,7 @@ Building the documentation requires several dependencies, all of which can be
 installed with Spack:
 
 * sphinx
+* sphinxcontrib-programoutput
```
Comment (adamjstewart) [inline lib/spack/docs/contribution_guide.rst]: A `py-sphinxcontrib-programoutput` package coming soon to a Spack near you: #4267.

**Comment by adamjstewart** [comment]
We can also remove the vendored `sphinx_rtd_theme` too if you want. I'm starting to understand why other developers recommend fixing the versions of test dependencies. 2 weeks in a row is too much.

**Comment by adamjstewart** [comment]
Whoo! Victory is mine! Take that, Travis.

**Comment by alalazo** [comment]
Sorry to press, but ping @tgamblin :smile:

**Comment by adamjstewart** [comment]
I'm going to make an executive decision and merge this so that we can use Travis again. Even if it introduces a bug, it's better than the current state of things 😆

**Comment by adamjstewart** [comment]
Restarted unit tests for the last 20 or so failing PRs.

**Pull request merged**